src/main/java/graphql/nadel/engine/NadelExecutionStrategy.java

```diff
@@ -223,16 +226,22 @@ private Service getServiceForFieldDefinition(GraphQLFieldDefinition fieldDefinit
         Map<Field, FieldTransformation> transformationByResultField = queryTransformerResult.getTransformationByResultField();
         List<MergedField> transformedMergedFields = queryTransformerResult.getTransformedMergedFields();
 
-        ServiceExecutionParameters serviceExecutionParameters = newDelegatedExecutionParameters()
-                .query(query)
-                .build();
+        ServiceExecutionParameters serviceExecutionParameters = buildServiceExecutionParameters(executionContext, query);
         ServiceExecution serviceExecution = service.getServiceExecution();
         GraphQLSchema underlyingSchema = service.getUnderlyingSchema();
 
         return serviceExecution.execute(serviceExecutionParameters)
-                .thenApply(delegatedExecutionResult -> resultToResultNode.resultToResultNode(context, delegatedExecutionResult, rootExecutionStepInfo, transformedMergedFields, underlyingSchema, operation))
+                .thenApply(delegatedExecutionResult -> resultToResultNode.resultToResultNode(executionContext, delegatedExecutionResult, rootExecutionStepInfo, transformedMergedFields, underlyingSchema, operation))
                 .thenApply(resultNode -> serviceResultNodesToOverallResult.convert(resultNode, overallSchema, transformationByResultField));
     }
 
-
+    private ServiceExecutionParameters buildServiceExecutionParameters(ExecutionContext executionContext, Document query) {
+        return newServiceExecutionParameters()
+                    .query(query)
+                    .context(executionContext.getContext())
+                    .variables(executionContext.getVariables())
```

Comment:
the challenge here is that only variables and fragments should be passed which are actually used in the transformed query.